pkg/transformers/builder/transformer_builder.go

```diff
@@ -18,6 +18,12 @@ type TransformerBuilder struct {
 
 type Option func(b *TransformerBuilder)
 
+type TransformerDefinition struct {
```

Comment:
I think this is a mix of two things. One, the definition of a transformer, and another the build function. I think this might be better off by separating the two concerns. The `TransformerDefinition` should probably move to the transformers root package (instead of the builder), and not contain the build function. 

```
type Definition struct {
	SupportedTypes []SupportedDataType
	Parameters     []TransformerParameter
}
```